packages/governance/test/swingsetTests/committeeBinary/vat-voter.js

```diff
@@ -0,0 +1,36 @@
+// @ts-check
+
+import { E } from '@agoric/eventual-send';
+import { Far } from '@agoric/marshal';
+
+const build = async (log, zoe) => {
+  return Far('voter', {
+    createVoter: async (name, voteFacet) => {
+      return Far(`Voter ${name}`, {
+        voteBallot: (qName, ballot) => {
+          log(`Voter ${name} cast a ballot for ${ballot.chosen}`);
+          return E(voteFacet).castBallot(ballot);
+        },
+        verifyBallot: async counterInstance => {
+          const publicFacet = E(zoe).getPublicFacet(counterInstance);
+          const ballotTemplate = E(publicFacet).getBallotTemplate();
+          const positionP = E(ballotTemplate).getPositions();
+          const qNameP = E(ballotTemplate).getQuestion();
+          const methodP = E(ballotTemplate).getMethod();
+          const [position, method, qName] = await Promise.all([
+            positionP,
+            methodP,
+            qNameP,
+          ]);
+          log(`Verify ballot from instance: ${qName}, ${position}, ${method}`);
```

Comment:
How is this verifying the ballot? does the voter have to request all of this information and compare each? That doesn't seem right. Why not have a WeakSet that contains all of the properly made ballots and validate by sending the ballot in to be checked against the WeakSet?

Reply:
We will want better support for usability. At this point, I'm trying to demonstrate that a voter can get a ballot and tell that it's connected to a particular BallotCounter and Registrar.

If you'd rather wait until everything is connected, I can stop sending PRs until all the pieces have been written.